forge-game/src/main/java/forge/game/GameAction.java

```diff
@@ -220,6 +220,9 @@ private Card changeZone(final Zone zoneFrom, Zone zoneTo, final Card c, Integer
         if (fromBattlefield && !toBattlefield) {
             c.getController().setRevolt(true);
         }
+        if (zoneTo.is(ZoneType.Graveyard) && c.isPermanent()) {
```

Comment:
also non Token

@tool4ever  @Agetian does this needs to be checked for Merged card parts?

Reply:
yea it would have to be
perhaps it would be cleanest to do this in `Zone.add`

Reply:
I moved it there!